.github/workflows/integrations_tests.yml

```diff
@@ -0,0 +1,94 @@
+name: integration tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to test on'
+        required: true
+      pytorch_nightly:
+        description: 'Whether to test integration tests (diffusers + transformers)'
+        required: false
+        default: false
+
+jobs:
+  run_transformers_integration_tests_main:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: "pip"
+          cache-dependency-path: "setup.py"
+      - name: print environment variables
+        run: |
+          echo "env.CI_BRANCH = ${{ env.CI_BRANCH }}"
+          echo "env.CI_SHA = ${{ env.CI_SHA }}"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install .[test]
+          cd .. && git clone https://github.com/huggingface/transformers.git && cd transformers/ && git log
+          python -m pip install ".[dev]"
+
+      - name: Test transformers integration
+        run: |
+          RUN_SLOW=1 pytest tests/peft_integration/test_peft_integration.py
+  run_transformers_integration_tests_release:
```

Comment:
How is it controlled that this uses the release, not the latest main? I couldn't find that in the config.

Also, there is a lot of duplication with the previous task, depending on how they differ, it might be better to turn this into a test matrix.

Reply:
> How is it controlled that this uses the release, not the latest main? I couldn't find that in the config.

for `run_transformers_integration_tests_release` I skip the install from main part of the workflow

Reply:
Ah I see, thx. So that should be pretty easy to integrate into a test matrix, WDYT?

Reply:
I think so yes, let me give it a try